phpBB/phpbb/db/driver/driver_interface.php

```diff
@@ -289,12 +289,37 @@ public function sql_fetchrow($query_id = false);
 	public function cast_expr_to_bigint($expression);
 
 	/**
-	* Get last inserted id after insert statement
-	*
-	* @return	string		Autoincrement value of the last inserted row
-	*/
+	 * Gets the ID of the **last** inserted row immediately after an INSERT
+	 * statement.
+	 *
+	 * **Note**: Despite the name, the returned ID refers to the row that has
+	 * just been inserted, rather than the hypothetical ID of the next row if a
+	 * new one was to be inserted.
+	 *
+	 * The returned value can be used for selecting the item that has just been
+	 * inserted or for updating another table with an ID pointing to that item.
+	 *
+	 * Will be deprecated in a future version of phpBB in favor of
+	 * `sql_last_inserted_id`.
```

Comment:
Is this OK to mention "Will be deprecated"? Signals that there's an eventual intention to deprecate, but without doing so immediately.

Reply:
I would actually add `@deprecated 3.3.8-RC1 Replaced by sql_last_inserted_id(), to be removed in 4.0.0-a1`